File: Ruby/6KYU/Guess.rb
class Guesser
  def get_number
    bot = 1; top = 1000
    (0..9).each do |x|
      num = ((top-bot)/2).round + bot
      a = guess(num);
      bot = num if a=="Too low!"
      top = num if a=="Too high!"
      return num if a=="Correct!"
    end
    return "Nope!"
  end
end

File: Javascript/6KYU/Guess.js
Guesser.prototype.getNumber = function() {
  bot = 1; top = 1000;
  for(myi=0;myi<10;myi++){
    num =Math.round((top-bot)/2) + bot
    a = this.guess(num);
    if(a==="Too low!"){bot = num;}
    if(a==="Too high!"){top = num;}
    if(a==="Correct!"){return num;}
  }
  return num;
}